bridge/contracts/Federation.sol

```diff
@@ -164,6 +230,42 @@ contract Federation is Initializable, UpgradableOwnable {
         return true;
     }
 
+  function acceptTransfer(
+    address originalTokenAddress,
+    address payable sender,
+    address payable receiver,
+    uint256 number,
+    bytes32 blockHash,
+    bytes32 transactionHash,
+    uint32 logIndex,
+    TokenType tokenType
+  ) internal {
+    if (tokenType == TokenType.NFT) {
+      require(bridgeNFT != INFTBridge(0), "Federation: Empty bridgeNFT");
+      bridgeNFT.acceptTransfer(
+        originalTokenAddress,
+        sender,
+        receiver,
+        number,
+        blockHash,
+        transactionHash,
+        logIndex
+      );
+      return;
+    }
+
+    // if it is not the NFT token type it should send as coin
```

Comment:
I'd maybe remove this, I think the code is already pretty clear (but keep it it you want of course)